code/__HELPERS/turfs.dm

```diff
@@ -376,6 +376,33 @@ Turf and target are separate in case you want to teleport some distance from a t
 		if (target)
 			return target
 
+///Returns a random department of areas to pass into get_safe_random_station_turf() for more equal spawning.
+/proc/get_safe_random_station_turf_equal_weight()
+	// Big list of departments, each with lists of each area subtype.
+	var/list/department_areas = list(
+				subtypesof(/area/station/engineering), \
+				subtypesof(/area/station/medical), \
+				subtypesof(/area/station/science), \
+				subtypesof(/area/station/security), \
+				subtypesof(/area/station/service), \
+				subtypesof(/area/station/command), \
+				subtypesof(/area/station/hallway), \
+				subtypesof(/area/station/ai_monitored), \
+				subtypesof(/area/station/cargo)
+			)
```

Comment:
that's a very heavy list, make it static (don't forget to move init behind an isnull check)